vesktop: add --enable-wayland-ime flag to NIXOS_OZONE_WL

LilleAila opened the pull request:
This pull request adds the `--enable-wayland-ime` flag to the `NIXOS_OZONE_WL` flags, so that IMEs work in vesktop on wayland.

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

LilleAila committed:
vesktop: wayland IME option

This commit adds an option to wrap the resulting program with the
`--enable-wayland-ime` argument, so that IME support on wayland can be
enabled in the package.

Aleksanaa commented on pkgs/by-name/ve/vesktop/package.nix:
```diff
@@ -128,6 +129,7 @@ stdenv.mkDerivation (finalAttrs: {
         --add-flags $out/opt/Vesktop/resources/app.asar \
         ${lib.optionalString withTTS "--add-flags \"--enable-speech-dispatcher\""} \
+        ${lib.optionalString withWaylandIME "--add-flags \"--enable-wayland-ime\""} \
         --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
```
Maybe add it here? Does `--enable-wayland-ime` bring any overhead to people not using an IME?

LilleAila replied:
The value is set to false by default, so users who do not want it should not get any extra overhead.

LilleAila replied:
I moved that flag to `NIXOS_OZONE_WL` and removed the option I created.

getchoo reviewed (changes requested):
https://github.com/NixOS/nixpkgs/pull/306310#discussion_r1576646996

Aleksanaa commented:
> [#306310 (comment)](https://github.com/NixOS/nixpkgs/pull/306310#discussion_r1576646996)

small note: when you comment on others' reviews, you are already reviewing it

LilleAila committed:
vesktop: IME with NIXOS_OZONE_WL

This moves the `--enable-wayland-ime` flag to NIXOS_OZONE_WL instead of
having a separate option for it.

vgskye reviewed (approved):
wait, IME didn't work with vesktop wayland? LGTM :3

pluiedev commented:
> wait, IME didn't work with vesktop wayland? LGTM :3

It in fact didn't and that annoyed me to no end :] Had to keep doing `vesktop --enable-wayland-ime &; disown %1` manually

pluiedev reviewed (commented):
Honestly... I prefer #294601's approach better where we unify what `NIXOS_OZONE_WL` does across all apps, including `--enable-wayland-ime`. This is okay as a stop-gap measure though

Also for future reference, perhaps the title and the description can be revised to reflect that no option has been added, and the PR is merely adding an additional flag
